client-linuxapp/src/serviceinfo.rs

```diff
@@ -69,6 +72,61 @@ fn create_user(user: &str) -> Result<()> {
     Ok(())
 }
 
+fn create_user_with_password(user: &str, password: &str) -> Result<()> {
+    // Checks if user already present
+    let user_info = passwd::Passwd::from_name(user);
+    if user_info.is_some() {
+        log::info!("User {} is already present", user);
+        return Ok(());
+    }
+    // Creates new user if user not present
+    log::info!("Encrypting password");
+    let pw_create = Command::new("echo")
+        .arg(format!("{}|openssl", password))
```

Comment:
In order to pipe one command output to another you need to use `Stdio::piped()` in the source and `.stdin(Stdio::from(x))` in the 'reader', but you need to do it in two steps, first the `echo` part of the command, where you are going to say that `stdout` is going to be piped; and then the `openssl` part as another command, where you are going to say that `stdin` comes from the first command:

```
let echo = Command::new("echo")
        .arg("my password")
        .stdout(Stdio::piped())
        .spawn()
        .expect("some error");
let openssl = Command::new("openssl")
        .arg("passwd")
        .arg("-1")
        .arg("-stdin")
        .stdin(Stdio::from(echo.stdout.expect("another error")))
        .output()
        .expect("yet another error");
```

Reply:
I think I understand, I'll try that out then. Going to attempt to use a rust or openssl crate first, that sounds like it might be a cleaner implementation.